src/block-experiments-setup.js

```diff
@@ -1,8 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { addAction } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
 // This file is to set up the jetpack/layout-grid block that currently lives in block-experiments/blocks/layout-grid
 import { registerBlock } from '../block-experiments/blocks/layout-grid/src';
 
-export default function setupBlockExperiments( capabilities ) {
-	if ( capabilities.layoutGridBlock ) {
-		registerBlock();
-	}
-}
+const setupHooks = () => {
+	// Hook triggered after the editor is rendered
+	addAction(
+		'native.render',
+		'gutenberg-mobile-block-experiments',
+		( props ) => {
+			const capabilities = props.capabilities ?? {};
+
+			// Register Layout Grid block
+			if ( capabilities.layoutGridBlock ) {
+				registerBlock();
+			}
+		}
+	);
+};
+
+export default () => {
+	setupHooks();
+};
```

Comment:
Further along in my review, I noticed this particular code structure is [already in place](https://github.com/wordpress-mobile/gutenberg-mobile/blob/82227ef9da1e238b566e04a370e923f545cbc347/src/setup.js#L49). Feel free to ignore my comment if the preference is to follow established structure.

Reply:
I agree that we could simplify this file by unifying the logic into the default function or an exported named function, my idea was to apply the same structure for the plugin setup files as GB-mobile, especially in order to encourage structuring a bit more strictly how plugins arrange their setup code. Probably it's not worthwhile today but I think it would be great to start exploring and inducing structure improvements for future plugin additions, wdyt?

Reply:
@fluiddot I agree it makes most sense to postpone any code restructuring. Thanks!